apacheconfig/wloader.py

```diff
@@ -0,0 +1,423 @@
+        """Returns the contents of this node as it would appear in a config
+        file.
+        """
+
+    @abc.abstractproperty
+    def ast_node_type(self):
+        """Returns object typestring as defined by the apacheconfig parser.
+        """
+
+    @abc.abstractproperty
+    def whitespace(self):
+        """Returns preceding or trailing whitespace for this node as a string.
+       """
+
+    @abc.abstractmethod
+    @whitespace.setter
+    def whitespace(self, value):
+        """Set preceding or trailing whitespace for this node.
+
+        Args:
+            value (str): value to set whitespace to.
+        """
+
+
+class ContentsNode(Node):
+    """Creates object representing an ordered list of BlockNodes and ItemNodes.
+
+    Each BlockNode contains a ContentsNode, and every configuration file's
+    root should be a ContentsNode. To construct from a raw string, use the
+    `parse` factory function. The default __init__ constructor expects data
+    from the internal apacheconfig parser.
+
+    Args:
+        raw (list): Raw data returned from ``apacheconfig.parser``.
+    """
+    def __init__(self, raw):
+        self._type = raw[0]
+        self._contents = []
+        self._whitespace = ""
+        for elem in raw[1:]:
+            if isinstance(elem, str) and elem.isspace():
+                self._whitespace = elem
+            elif elem[0] == "block":
+                self._contents.append(BlockNode(elem))
+            else:
+                self._contents.append(ItemNode(elem))
+
+    @staticmethod
+    def parse(raw_str, options={}, parser=None):
+        """Factory for :class:`apacheconfig.ContentsNode` by parsing data from
+        config string.
+
+        Args:
+            raw_str (str): Config string to parse.
+            options (dict): Additional options to pass to the created parser.
+                Ignored if another ``parser`` is supplied.
+            parser (:class:`apacheconfig.ApacheConfigParser`): optional, to
+                re-use an existing parser. If ``None``, creates a new one.
+        Returns:
+            :class:`apacheconfig.ContentsNode` containing data parsed from
+            ``raw_str``.
+        """
+        if not parser:
+            parser = _create_apache_parser(options, start='contents')
+        return ContentsNode(parser.parse(raw_str))
+
+    def add(self, index, raw_str):
+        """Parses given string into appropriate :class:`apacheconfig.Node`
+        object, then adds to contents at specified index.
+
+        Args:
+            index (int): index of contents at which to insert the node.
+            raw_str (str): string to parse. The parser will automatically
+                determine whether it's a :class:`apacheconfig.BlockNode` or
+                :class:`apacheconfig.ItemNode`.
+
+        Returns:
+            The :class:`apacheconfig.Node` created from parsing ``raw_str``.
+        """
+        parser = _create_apache_parser({}, start='miditem')
```

Comment:
May be this is an outdated concern, so just not to miss it - recreating the whole parser suite may be expensive!